### PR TITLE
Resolve error groups

### PR DESCRIPTION
**Pull request by t-romani**
Solves #55 
Now an error group can get _resolved_

Resolved error tab doesn't show resolve button

Screen Shot 2020-06-01 at 15 40 14

### CODE REVIEW

File: spec/dummy/db/migrate/20200601134029_create_exception_hunter_error_groups.rb [hunk 1]
```diff
@@ -5,6 +5,7 @@ def change
     create_table :exception_hunter_error_groups do |t|
       t.string :error_class_name, null: false
       t.string :message
+      t.integer :status, default: 0
```
Comment (brunvez) [inline spec/dummy/db/migrate/20200601134029_create_exception_hunter_error_groups.rb:8]: We are using this field in lots of queries, it'd be good to index by it

File: app/models/exception_hunter/error.rb [hunk 2]
```diff
@@ -6,6 +6,7 @@ class Error < ApplicationRecord
     belongs_to :error_group, touch: true
 
     before_validation :set_occurred_at, on: :create
+    after_create :unresolve_error_group
```
[the remark this answers is not in the document]
Comment (brunvez) [reply app/models/exception_hunter/error.rb:9]: Tests?

File: app/models/exception_hunter/error.rb [hunk 3]
```diff
@@ -18,11 +19,23 @@ class Error < ApplicationRecord
     scope :in_current_month, lambda {
       in_period(Date.current.beginning_of_month.beginning_of_day..Date.current.end_of_month.end_of_day)
     }
+    scope :from_active_error_groups, lambda {
```
Comment (brunvez) [inline app/models/exception_hunter/error.rb:22]: Tests?

File: app/models/exception_hunter/error.rb [hunk 4]
```diff
@@ -18,11 +19,23 @@ class Error < ApplicationRecord
...
+      joins(:error_group).where(error_group_id: ErrorGroup.active)
+    }
+    scope :from_resolved_error_groups, lambda {
```
Comment (brunvez) [inline app/models/exception_hunter/error.rb:25]: Tests?

File: app/models/exception_hunter/error_group.rb [hunk 5]
```diff
@@ -21,7 +23,8 @@ class ErrorGroup < ApplicationRecord
         .where(Error.in_last_7_days.where(Error[:error_group_id].eq(ErrorGroup[:id])).arel.exists)
     }
     scope :with_errors_in_current_month, lambda {
-      joins(:grouped_errors).where(Error.in_current_month.where(Error[:error_group_id].eq(ErrorGroup[:id])).arel.exists)
+      joins(:grouped_errors)
+        .where(Error.in_current_month.where(Error[:error_group_id].eq(ErrorGroup[:id])).arel.exists)
```
[the remark this answers is not in the document]
Comment (t-romani) [reply app/models/exception_hunter/error_group.rb:27]: just a newline 😆

**Comment by t-romani** [comment]
missing button css ⏹️

File: .reek.yml [hunk 6]
```diff
@@ -32,6 +32,7 @@ detectors:
     exclude:
       - ExceptionHunter::Middleware::RequestHunter#catch_prey
       - ExceptionHunter::Middleware::SidekiqHunter#track_exception
+      - ExceptionHunter::DashboardPresenter#calculate_tabs_counts
```
Comment (brunvez) [inline .reek.yml:35]: For once, the dude is right. We’ll fix it in a refactor

File: app/controllers/exception_hunter/resolved_errors_controller.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+module ExceptionHunter
+  class ResolvedErrorsController < ApplicationController
+    def create
+      ErrorGroup.find(resolved_error_params[:id]).resolved!
+
+      redirect_to errors_path, notice: 'Error resolved successfully'
+    end
+
+    def resolved_error_params
```
Comment (brunvez) [inline app/controllers/exception_hunter/resolved_errors_controller.rb:11]: No need to use this for a find

File: spec/requests/exception_hunter/errors_request_spec.rb [hunk 8]
```diff
@@ -183,4 +183,21 @@ module ExceptionHunter
       end
     end
   end
+
+  describe 'resolve' do
```
Comment (brunvez) [inline spec/requests/exception_hunter/errors_request_spec.rb:187]: Use the route instead of the method

**Pull request merged**